### PR TITLE
Add docs regarding istio patch

### PR DESCRIPTION
**Pull request by sjanota**
**Description**

Add documentation describing how we patch istio and how to use kyma with your own istio.

**Related issue(s)**

### CODE REVIEW

File: docs/service-mesh/docs/100-istio-installation.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+* All resources from charts `prometheus`, `tracing`, `grafana` and `servicegraph` are deleted
+
+You can find more details about those changes in component `istio-kyma-patch`.
+
+## Using your own Istio
+
+It is possible to install Kyma with Istio already installed. If you have running kubernetes with istio on it you can 
+add Kyma to the cluster.
+
+>**NOTE:** Bear in mind, that patch will always be applied and will change existing Istio installation as described
```
[the remark this answers is not in the document]
Comment (sjanota) [reply docs/service-mesh/docs/100-istio-installation.md]: Well "the rest" is the reason why we have this documentation? To describe how to get Kyma running with preinstalled istio.

File: docs/service-mesh/docs/100-istio-installation.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+By default Istio is installed during Kyma installation. The process is split into two parts:
+1. Installation of official istio charts. Charts are vendored within Kyma repository under `resources/istio`. 
+2. Patch of Istio to make it suitable for Kyma. 
+
+During installation raw official charts from currently supported Istio release are installed on cluster. The only 
+customization done at this point of installation is enabling security in Istio. This is prerequisite to run Kyma and 
+every Istio installation for Kyma needs to have security enabled.
+ 
+Second stage, patch is done by kubernetes job which introduces following changes into Istio:
```
[the remark this answers is not in the document]
Comment (sjanota) [reply docs/service-mesh/docs/100-istio-installation.md]: It's important part. Describes what changes are applied to istio so user is aware in which way we're messing with his Istio.

File: docs/service-mesh/docs/100-istio-installation.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+2. A custom Istio Patch is applied to further customize the Istio installation. A Kubernetes job introduces these 
+changes:
+  - A memory limit is set for every sidecar
+  - Istio components use Zipkin in the `kyma-system` Namespace, instead of the default `istio-system`
+  - A webhook is added to the Istio Pilot
+  - A TLS certificate is created for the Ingress Gateway
+  - All resources related to the `prometheus`, `tracing`, `grafana`, and `servicegraph`charts  are deleted
+
+During installation raw official charts from currently supported Istio release are installed on cluster. The only
```
Comment (tomekpapiernik) [inline docs/service-mesh/docs/100-istio-installation.md]: Remove lines 20-22 - the same information is contained in lines 8-10

File: docs/service-mesh/docs/100-istio-installation.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+## Use an existing Istio installation with Kyma
+
+You can use an existing installation of Istio running on Kubernetes with Kyma. The custom Istio patch is applied to such 
+an installation.
+
+>**NOTE:** You cannot skip applying the Istio patch in the Kyma installation process.
+
+To allow such an implementation, you must install Kyma without Istio. To achieve this, modify the Installation custom 
+resource used to trigger the Kyma installer. Read the Installation document in the Custom Resource section of the Kyma
```
Comment (tomekpapiernik) [inline docs/service-mesh/docs/100-istio-installation.md]:
```suggestion
resource used to trigger the Kyma installer. Read the **Installation** CR document in the **Kyma** 
```

Comment (tomekpapiernik) [inline docs/service-mesh/docs/100-istio-installation.md]: Installation custom 
resource used to trigger ->

 Installation custom 
resource (CR) used to trigger

File: docs/service-mesh/docs/100-istio-installation.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  - Istio components use Zipkin in the `kyma-system` Namespace, instead of the default `istio-system`
+  - A webhook is added to the Istio Pilot
+  - A TLS certificate is created for the Ingress Gateway
+  - All resources related to the `prometheus`, `tracing`, `grafana`, and `servicegraph`charts  are deleted
+
+During installation raw official charts from currently supported Istio release are installed on cluster. The only 
+customization done at this point of installation is enabling security in Istio. This is prerequisite to run Kyma and 
+every Istio installation for Kyma needs to have security enabled.
+
+You can find more details about those changes in component `istio-kyma-patch`.
```
Comment (tomekpapiernik) [inline docs/service-mesh/docs/100-istio-installation.md]:
```suggestion
To learn more about the custom Istio patch applied in Kyma, see the `components/istio-kyma-patch/` directory. 
```

File: docs/service-mesh/docs/100-istio-installation.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
+---
+title: Istio installation
+type: Details
+---
+
+As a core component, Istio installs with every Kyma deployment by default. The installation consists of two steps:
+
+1. Istio installs using the official, raw charts from the currently supported release. The charts that are currently 
+used are stored in the resources/istio directory. The installation is customized by enabling security in Istio.
```
Comment (tomekpapiernik) [inline docs/service-mesh/docs/100-istio-installation.md]:
```suggestion
used are stored in the `resources/istio` directory. The installation is customized by enabling security in Istio.
```

File: docs/service-mesh/docs/100-istio-installation.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    >**NOTE:** Every installation of Istio for Kyma must have security enabled.
+
+2. A custom Istio Patch is applied to further customize the Istio installation. A Kubernetes job introduces these
```
Comment (tomekpapiernik) [inline docs/service-mesh/docs/100-istio-installation.md]:
```suggestion
2. A custom Istio patch is applied to further customize the Istio installation. A Kubernetes job introduces these 
```

**Pull request merged**